### ISSUE TITLE
Allow breakdown table columns to be specified

### ISSUE THREAD

**Issue by aliscott** [infracost/infracost#524]
In v0.8.0 we simplified the breakdown table view by removing some default columns. We can now look at adding an option to specify the output columns.

a) Add a `--fields` flag to the `infracost breakdown` that takes the following as a string array argument. The output order should always be the following. Output a warning if the format is not table. If fields is empty, use the defaults (name, monthly_quantity, unit, monthly_cost).
  * name (always here, users should not be allowed to pass this in)
  * price
  * monthly_quantity
  * unit
  * hourly_cost
  * monthly_cost

b) Add the same functionality to `--fields` flag to the `infracost output` command. 

c) Ignore the columns for usage-based resources and show the existing messaging. This would also mean for the first cut the columns should always be in the same consistent order.

-----

**Previous description**: which is out of scope for this GH issue:

I think the "name" column shouldn't be optional, so it always appears.

**Q1:** `--fields` vs `--columns` vs other? I'm leaning towards `--fields` over `--columns` because then it can also be used with the `infracost diff` command in the future.

If `infracost breakdown` is run for non-table formats and the flag is specified we should show a warning but not error.

**Output command**

We should have the same flag and functionality for the `infracost output` command with the same warnings if non-table formats are specified.

**How should this work with usage-based resources?**

For usage-based resources without costs we currently show the following `Cost depends on usage: $X per Y` spanned across multiple columns. In the future we could show or hide this depending on which columns they have selected, but as a first cut I think we should keep this functionality as is, i.e. ignore the columns for usage-based resources and show the existing messaging. This would also mean for the first cut the columns should always be in the same consistent order.

**Q2:** Any thoughts on the above and how we handle it going forward?

**HTML output**

**Q3:** Should we add this as part of the HTML output in the first cut?

**Config file**

**Q4:** Should we add this to the Infracost config file in the first cut? If so, what should the config file key be - would we want a separate config file key for diff and table formats going forward?

**Comment by sinabakh** [comment]
A nice feature to have 🚀 

Q1: I also prefer `fields`.
Q2: I got no concrete idea.
Q3: I believe we have to support the HTML output but for starters, the table should be enough. 
Q4: I think so. I would prefer to configure my config file instead of passing 2 lines of arguments to the cmd.

**Comment by aliscott** [comment]
If we add JS to the page then another option for Q3 is to allow the columns to be shown/hidden on the HTML page itself.

**Comment by ekifel** [comment]
@alikhajeh1 Hey, I want to work on it!

**Comment by alikhajeh1** [comment]
@ekifel any update on this? plz lmk if I can help :)

**Comment by ekifel** [comment]
> @ekifel any update on this? plz lmk if I can help :)

Oops, forgot about it, my bad. Sent PR on review.

**Comment by alikhajeh1** [comment]
@dawidmalina, @lcbm and @sinabakh just FYI that `--fields` was released in v0.8.6, it works with `infracost breakdown` and `infracost output` with the table output format, the HTML format is being updated as part of https://github.com/infracost/infracost/issues/548:

```
--fields strings   Comma separated list of output fields: price,monthlyQuantity,unit,hourlyCost,monthlyCost.
                   Only supported by table output format (default [monthlyQuantity,unit,monthlyCost])
```
